dev/lint-repository.sh

```diff
@@ -22,14 +22,13 @@ then
     # can still check that they don't worsen.
     CUR_HEAD=${TRAVIS_COMMIT_RANGE%%...*}
     PR_HEAD=${TRAVIS_COMMIT_RANGE##*...}
-    MERGE_BASE=$(git merge-base $CUR_HEAD $PR_HEAD)
-    dev/lint-commits.sh $MERGE_BASE $PR_HEAD || CODE=1
+    MERGE_BASE=$(git merge-base "$CUR_HEAD" "$PR_HEAD")
+    dev/lint-commits.sh "$MERGE_BASE" "$PR_HEAD" || CODE=1
 fi
 
 # Check that the files with 'whitespace' gitattribute end in a newline.
 # xargs exit status is 123 if any file failed the test
-find . "(" -path ./.git -prune ")" -type f \
--o "(" -exec dev/tools/should-check-whitespace.sh '{}' ';' ")" \
--print0 | xargs -0 -L 1 dev/tools/check-eof-newline.sh || CODE=1
+find . "(" -path ./.git -prune ")" -o -type f -print0 |
+    xargs -0 dev/tools/check-eof-newline.sh || CODE=1
```

Comment:
Previous version removed the xargs which broke the test since find doesn't fail when -exec does. This is now fixed.